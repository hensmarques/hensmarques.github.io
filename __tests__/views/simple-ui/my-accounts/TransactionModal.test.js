import React from 'react';
import { TransactionModal } from '../../../../src/views/simple-ui/my-accounts/TransactionModal';

describe('<TransactionModal/>', () => {
  const mockCurrency = {
    id: 'BTC',
    type: 'crypto',
  };
  const loadUserSettings = jest.fn();

  it('renders with transactionType deposit and matches the snapshot', () => {
    const wrapper = shallow(
      <TransactionModal
        currency={mockCurrency}
        loadUserSettings={loadUserSettings}
        transactionType="deposit"
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('renders with transactionType withdraw', () => {
    const wrapper = shallow(
      <TransactionModal
        currency={mockCurrency}
        loadUserSettings={loadUserSettings}
        transactionType="withdraw"
      />,
    );
  });

  it("dispatches showModal('none') when modal closes", () => {
    const showModal = jest.fn();
    const wrapper = shallow(
      <TransactionModal
        currency={mockCurrency}
        loadUserSettings={loadUserSettings}
        showModal={showModal}
        transactionType="deposit"
      />,
    );
    wrapper
      .find('Connect(Modal)')
      .props()
      .onClose();
    expect(showModal).toHaveBeenCalledWith('none');
  });
});
