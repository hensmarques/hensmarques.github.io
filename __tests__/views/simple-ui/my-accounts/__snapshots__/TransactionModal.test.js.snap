// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TransactionModal/> renders with transactionType deposit and matches the snapshot 1`] = `
<Connect(Modal)
  className="deposit-crypto"
  onClose={[Function]}
>
  <h3>
    Two-factor authentication is required for making deposits and withdraws.
  </h3>
  <p>
    <Link
      replace={false}
      to="/settings"
    >
      Click here to change your 2FA settings.
    </Link>
  </p>
</Connect(Modal)>
`;
